misc/atrium-specs/src/commonMain/kotlin/ch/tutteli/atrium/specs/integration/AbstractIterableNotToContainEntriesExpectationsTest.kt

```diff
@@ -0,0 +1,294 @@
+import ch.tutteli.atrium.specs.integration.utils.SubjectLessTestData
+import ch.tutteli.atrium.specs.invoke
+import ch.tutteli.atrium.specs.listBulletPoint
+import ch.tutteli.atrium.specs.name
+import ch.tutteli.atrium.specs.rootBulletPoint
+import ch.tutteli.atrium.specs.toEqualDescr
+import ch.tutteli.atrium.specs.warningBulletPoint
+import ch.tutteli.atrium.testfactories.TestFactory
+import ch.tutteli.atrium.translations.DescriptionComparableExpectation
+import ch.tutteli.atrium.translations.DescriptionIterableLikeExpectation
+
+
+@Suppress("FunctionName")
+abstract class AbstractIterableNotToContainEntriesExpectationsTest(
+    private val notToContainEntriesSpec: Fun2<Iterable<Double>, Expect<Double>.() -> Unit, Array<out Expect<Double>.() -> Unit>>,
+    private val notToContainNullableEntriesSpec: Fun2<Iterable<Double?>, (Expect<Double>.() -> Unit)?, Array<out (Expect<Double>.() -> Unit)?>>,
+    private val notToHaveElementsOrNoneFunNameSpec: String,
+) : ExpectationFunctionBaseTest() {
+
+    @TestFactory
+    fun subjectLessTest() = subjectLessTestFactory(
+        SubjectLessTestData(
+            notToContainEntriesSpec.forSubjectLessTest({ toEqual(2.3) }, arrayOf())
+        ),
+        SubjectLessTestData(
+            notToContainNullableEntriesSpec.forSubjectLessTest({ toEqual(2.3) }, arrayOf())
+        )
+    )
+
+    private val notToContainDescr = DescriptionIterableLikeExpectation.NOT_TO_CONTAIN.getDefault()
+    private val toBeLessThanDescr = DescriptionComparableExpectation.TO_BE_LESS_THAN.getDefault()
+    private val toBeGreaterThanDescr = DescriptionComparableExpectation.TO_BE_GREATER_THAN.getDefault()
+
+    @TestFactory
+    fun expectationCreatorTest() = expectationCreatorTestFactory(
+        ExpectationCreatorTestData(
+            oneToSeven().toList().asIterable(),
+            expectationCreator = ExpectationCreatorTriple(
+                "${notToContainEntriesSpec.name} - first empty",
+                "$toBeGreaterThanDescr: 8.0",
+                {
+                    notToContainEntriesSpec.invoke(
+                        this,
+                        { toBeGreaterThan(8.0) },
+                        arrayOf(expectLambda { toBeGreaterThan(10.0) })
+                    )
+                },
+                { notToContainEntriesSpec.invoke(this, {}, arrayOf(expectLambda { toBeGreaterThan(10.0) })) }
+            ),
+            ExpectationCreatorTriple(
+                "${notToContainEntriesSpec.name} - second empty",
+                "$toBeGreaterThanDescr: 10.0",
+                {
+                    notToContainEntriesSpec.invoke(
+                        this,
+                        { toBeGreaterThan(8.0) },
+                        arrayOf(expectLambda { toBeGreaterThan(10.0) })
+                    )
+                },
+                { notToContainEntriesSpec.invoke(this, { toBeGreaterThan(8.0) }, arrayOf(expectLambda<Double> {})) }
+            )
+        ),
```

Comment:
You can re-use the old logic together with toVararg (same same for notToContainNullableEntriesSpec):
```suggestion
        notToContainEntriesSpec.forExpectationCreatorTest(
            "${IterableToContainEntriesSpecBase.Companion.toBeGreaterThanDescr}: 8.0",
            "${IterableToContainEntriesSpecBase.Companion.toBeGreaterThanDescr}: 10.0",
            { toBeGreaterThan(8.0) }, arrayOf(expectLambda { toBeGreaterThan(10.0) })
        ).toVararg().let { (first, rest) ->
            ExpectationCreatorTestData(oneToSeven().toList().asIterable(), first, *rest)
        },
```

Reply:
Although I agree, it makes the code look non-uniform and doesn't follow the convention used in the other migrations. I would rather some of these fields be extracted out to variables which can be reused across different test data. Does that work?

Reply:
I am not sure what you are talking about, forExpectationCreatorTest is the norm (see usage of AssertionCreatorSpec, forExpectationCreatorTest is almost always used). But maybe you mean `${IterableToContainEntriesSpecBase.Companion.toBeGreaterThanDescr}` that's a mistake on my part (quickly checked if it works in Intellij, and it looks like intellij modified some imports). You can of course write:
```suggestion
notToContainEntriesSpec.forExpectationCreatorTest(
            "$toBeGreaterThanDescr: 8.0",
            "$toBeGreaterThanDescr: 10.0",
            { toBeGreaterThan(8.0) }, arrayOf(expectLambda { toBeGreaterThan(10.0) })
        ).toVararg().let { (first, rest) ->
            ExpectationCreatorTestData(oneToSeven().toList().asIterable(), first, *rest)
        },
 ```